cdragontoolbox/export.py

```diff
@@ -659,3 +662,31 @@ def convert(self, fin, output, path):
                 name = os.path.join(obj_output_path, entry["name"] + ".obj")
                 with open(name, "w") as f:
                     f.write(sknfile.to_obj(entry))
+
+class RstConverter(FileConverter):
+    def __init__(self, regex):
+        self.regex = regex
+        self.hashes = HashFile(os.path.join(os.path.dirname(__file__), "hashes.rst.txt"), hash_size=16).load()
+
+    def is_handled(self, path):
+        return self.regex.search(path) is not None
+
+    def converted_paths(self, path):
+        yield path + '.json'
+
+    def convert(self, fin, output, path):
+        output_path = os.path.join(output, path + '.json')
+        rstfile = RstFile(fin)
+        bit_mask = (1 << rstfile.hash_bits) - 1
+        hashes = {hash & bit_mask: value for hash, value in self.hashes.items()}
+        rst_json = {}
+        if rstfile.font_config:
+            rst_json["__fontconfig"] = rstfile.font_config
+        for key, value in rstfile.entries.items():
+            if key in hashes:
+                key = hashes[key]
+            rst_json[key] = value
+        rst_json["__version"] = rstfile.version
+
+        with write_file_or_remove(output_path) as fout:
+            fout.write(json.dumps(rst_json, ensure_ascii=False).encode('utf-8'))
```

Comment:
To avoid superfluous conversions, open the file as binary:
```py
        with write_file_or_remove(output_path, True) as fout:
            fout.write(json.dumps(rst_json, ensure_ascii=False))
```

Reply:
Actually looking at it, it seems binary mode is the default and setting it to `False` could save me the `.encode('utf-8')`. Although that is working for me, I'm not sure it's save in regards to locales and stuff because I have no idea what happens when a file gets written to in non-binary mode...

Reply:
Oh sorry, I forgot it was the default and read your code backwards. :(

Using `False` should be a bit better as it will encode on-the-fly in the `write()` call instead of encoding the whole string at once.
But more importantly, it avoids juggling with encoding in this part of the code, which is cleaner.